src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1070,16 +1070,16 @@ private void prepareToJoin() throws ConfigurationException
         {
             Map<ApplicationState, VersionedValue> appStates = new EnumMap<>(ApplicationState.class);
 
-            if (SystemKeyspace.wasDecommissioned())
+            if (SystemKeyspace.wasDecommissioned() || SystemKeyspace.hasDecommissionFailed())
             {
                 if (OVERRIDE_DECOMMISSION.getBoolean())
                 {
-                    logger.warn("This node was decommissioned, but overriding by operator request.");
+                    logger.warn("This node was decommissioned or its decommission has failed, but overriding by operator request.");
```

Comment:
Here you could easily make the warn message text actually reflect if it was a failed decommission or the other